File: python/easy/find_first_right_target.py

def find_first_right_target(arr, target):
	result = (-1, -1)
	for i, item in enumerate(arr):
		if item == target:
			new_tuple = (i, item)
			result = new_tuple
	return result
print(find_first_right_target([1,2,3,1,2,3],3))
# O(n)

File: js/easy/findFirstRightTarget.js
function findFirstRightTarget(arr, target) {
  const result = [-1, -1];
  for (let i = 0; i < arr.length; i++) {
    if (arr[i] === target) {
      result[0] = i;
      result[1] = arr[i];
    }
  }
  return result;
}
console.log(findFirstRightTarget([1, 2, 3, 1, 2, 3], 3));
// O(n)

function findFirstRightTarget1(arr, target) {
  const result = [-1, -1];
  for (let i = arr.length; i > 0; i--) {
    if (arr[i] === target) {
      result[0] = i;
      result[1] = arr[i];
      break;
    }
  }
  return result;
}
console.log(findFirstRightTarget1([1, 2, 3, 1, 2, 3], 3));
// O(n-k)
